marketaccess/views.py

```diff
@@ -63,8 +74,27 @@ def get_context_data(self, form, **kwargs):
         if self.steps.current == self.SUMMARY:
             data = self.get_all_cleaned_data()
             context['all_cleaned_data'] = data
+        if form.errors:
+            for field in form:
+                context['formatted_form_errors'] = render_to_string(
+                    'marketaccess/report_barrier_form/error-link-list.html',
+                    {'form': form}
+                )
         return context
 
+    def render_next_step(self, form, **kwargs):
+        """
+        When using the NamedUrlWizardView, we have to redirect to update the
+        browser's URL to match the shown step.
+        """
+        status = self.get_cleaned_data_for_step(self.CURRENT_STATUS)['status']
+        if self.steps.current == self.CURRENT_STATUS and status != "4":
+            return redirect('market-access-emergency')
+        else:
+            next_step = self.get_next_step()
```

Comment:
these three lines can be replaced with `return super().render_next_step(form=form, **kwargs)`